src/Microsoft.AspNet.MiddlewareAnalysis/AnalysisServiceCollectionExtensions.cs

```diff
@@ -0,0 +1,20 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.AspNet.Hosting.Startup;
+using Microsoft.AspNet.MiddlewareAnalysis;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.DependencyInjection.Extensions;
+
+namespace Microsoft.Extensions.DependencyInjection
+{
+    public static class AnalysisServiceCollectionExtensions
+    {
+        public static IServiceCollection AddMiddlewareAnalysis(this IServiceCollection services)
+        {
+            // This should prevent AnalysisStartupFilter from being registered more than once.
+            services.TryAddEnumerable(ServiceDescriptor.Singleton<IStartupFilter, AnalysisStartupFilter>());
```

Comment:
Minor suggestion - might be better as transient - othewise the `AnalysisStartupFilter` instance will be added to the application scope service dictionary for the lifetime of the web app